pages/api/configuration.ts

```diff
@@ -5,83 +5,127 @@ import { domainMiddleware, jwtVerifyMiddleware } from "../../lib/middlewares";
 import MiddlewareError from "../../utils/MiddlewareError";
 import { getAuthToken } from "../../lib/environment";
 import {
-  FetchAppMetadataDocument,
-  FetchAppMetadataQuery,
   UpdateAppMetadataDocument,
   UpdateAppMetadataMutation,
-  MetadataItem,
-  MetadataInput,
+  FetchAppMetafieldsQuery,
+  FetchAppMetafieldsDocument,
+  FetchAppMetafieldsQueryVariables,
+  UpdateAppMetadataMutationVariables,
 } from "../../generated/graphql";
-
-const CONFIGURATION_KEYS = [
-  "THAT_SECRET",
-  "THIS_VALUE",
-];
-
-const prepareMetadataFromRequest = (input: MetadataInput[] | MetadataItem[]) =>
-  input
-    .filter(({ key }) => CONFIGURATION_KEYS.includes(key))
-    .map(({ key, value }) => ({ key, value }));
-
-const prepareResponseFromMetadata = (input: MetadataItem[]) => {
-  const output: MetadataInput[] = [];
-  for (const configurationKey of CONFIGURATION_KEYS) {
-    output.push(
-      input.find(({ key }) => key === configurationKey) ?? { key: configurationKey, value: "" }
-    );
-  }
-  return output.map(({ key, value }) => ({ key, value }));
-};
+import {
+  prepareMetadataFromRequest,
+  prepareResponseFromMetadata,
+} from "@/backend/configuration";
 
 const handler: NextApiHandler = async (request, response) => {
   let saleorDomain: string;
 
   try {
     saleorDomain = domainMiddleware(request) as string;
     await jwtVerifyMiddleware(request);
-  }
-  catch (e: unknown) {
+  } catch (e: unknown) {
     const error = e as MiddlewareError;
 
     console.error(error);
```

Comment:
intentional? if so I'd add some comment to indicate it's supposed to stay

Reply:
It's in api so I think it's fine here